libcaf_core/caf/telemetry/label.hpp

```diff
@@ -53,7 +53,15 @@ class CAF_CORE_EXPORT label : detail::comparable<label> {
 
   // -- comparison -------------------------------------------------------------
 
-  int compare(const label& x) const noexcept;
+  template <class T1, class T2>
+  static int compare(const T1& lhs, const T2& rhs) noexcept {
+    auto cmp1 = lhs.name().compare(rhs.name());
+    return cmp1 != 0 ? cmp1 : lhs.value().compare(rhs.value());
+  }
```

Comment:
Could we move this implementation as a free function to the cpp file? The intent being to minimize overhead when including the hpp.

Reply:
Unfortunately not, we need it from `label.cpp` and from `label_view.cpp`.

Reply:
Too bad, still nice refactoring.